weather_mv/loader_pipeline/pipeline.py

```diff
@@ -41,12 +44,75 @@ def pattern_to_uris(match_pattern: str) -> t.Iterable[str]:
         yield from [x.path for x in match.metadata_list]
 
 
+def validate_region(output_table: str, temp_location: t.Optional[str] = None, region: t.Optional[str] = None) -> None:
+    """Validates non-compatible regions scenarios by performing sanity check."""
+
+    def do_cleanup(bigquery_client: bigquery.Client, storage_client: storage.Client, dummy_output_table: str,
+                   dummy_bucket_name: str) -> None:
+        bigquery_client.delete_table(dummy_output_table, not_found_ok=True)
+        try:
+            storage_client.get_bucket(dummy_bucket_name).delete(force=True)
+        except NotFound:
+            pass
+
+    if not region and not temp_location:
+        raise ValueError('Invalid GCS location: None.')
+
+    bigquery_client = bigquery.Client()
+    storage_client = storage.Client()
+
+    bucket_region = region
+    table_region = None
+
+    if temp_location:
+        bucket = temp_location.split('gs://')
+        if len(bucket) < 2:
+            raise ValueError(f'Invalid GCS location: \'{bucket[0]}\'.')
+        bucket_name = bucket[1].split('/')[0]
+        bucket_region = storage_client.get_bucket(bucket_name).location
```

Comment:
I recommend using urlparse: https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlparse

It's a bit safer to use than re-implementing a parser, since it probably handles corner cases that we can't anticipate.

Reply:
Further, I expect it to throw better error messages.

Reply:
Necessary code changes have been done.